dashboard/app/models/user_level.rb

```diff
@@ -138,15 +130,31 @@ def before_unsubmit
     end
   end
 
-  def has_autolocked?(stage)
-    return false unless stage.lockable?
-    unlocked_at && unlocked_at < AUTOLOCK_PERIOD.ago
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
+  # automatically return `true` after `AUTOLOCK_PERIOD`, so the following
+  # are its getter and setter
+  def locked
+    unlocked_at.nil? || unlocked_at < AUTOLOCK_PERIOD.ago
   end
 
-  def locked?(stage)
+  def locked=(is_locked)
+    self.unlocked_at = is_locked ? nil : Time.now
+  end
+
+  # this is the "locked" value we return to the client. if the lesson isn't
+  # lockable or we are an authorized teacher, we always return `false`.
+  def show_as_locked?(stage)
```

Comment:
While we're updating this method, can we rename `stage` to `lesson`?

Reply:
(discussed offline- leaving for now)